src/App.js

```diff
@@ -51,6 +52,14 @@ function App() {
   const [state, dispatch] = useReducer(reducer, initialState);
 
   useEffect(() => {
+    if (!state.userCoordinates) {
+      const cachedCoordinates = getUserCoordinatesFromLocalStorage();
+
+      if (cachedCoordinates) {
+        dispatch({ type: 'setUserCoordinates', payload: cachedCoordinates })
+      }
+    }
```

Comment:
This useEffect is getting really busy, which makes it hard to maintain and bug prone. 
Can we separate this code to a separate use effect that will be called on mount?